File: apps/desktop-app/lib/src/api/dtos/snapshot_stats.dart
class SnapshotStats {
  int fileSize;
  DateTime createdAt;
  DateTime importedAt;

  SnapshotStats(this.fileSize, this.createdAt, this.importedAt);
  SnapshotStats.fromJSON(Map<String, dynamic> json)
      : fileSize = json['fileSize'],
        createdAt = DateTime.parse(json['createdAt']),
        importedAt = DateTime.parse(json['importedAt']);
}

File: libs/database/src/entities/snapshotStats.ts
import { Column, Entity, PrimaryGeneratedColumn } from "typeorm";

@Entity({ name: "snapshot_stats" })
export class SnapshotStats {
  @PrimaryGeneratedColumn()
  id!: number;
  @Column({ type: "integer" })
  fileSize!: number;
  @Column({ type: "datetime" })
  createdAt!: Date;
  @Column({ type: "datetime" })
  importedAt!: Date;
}
